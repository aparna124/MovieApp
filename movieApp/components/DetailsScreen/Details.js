import React from 'react';
import { StatusBar, StyleSheet, Text, View, Button, TouchableOpacity } from 'react-native';
import { Appbar} from 'react-native-paper';
import Card from '../HomeScreen/Card';
import { Feather } from '@expo/vector-icons';

const Details = ({route, navigation}) => {
    const {movieId, otherParams} = route.params;
    console.log("movie id is", movieId);
    console.log("Other params", otherParams);
    // currentPosterPath = otherParams.poster_path

    const Separator = () => (
        <View style={styles.separator} />
    );

    return (
        <View> 
            <View style={styles.topBarStyle}>
                <Text style={styles.titleStyle}>{otherParams.original_title}</Text>
            </View>
            <View style={{padding: '5%'}}>
                <View style={styles.movieDetailStyle}>

                    <Card key={movieId} {...otherParams} />

                    <View style={{padding: 10}}>
                        <Text style={styles.releaseDateTextStyle}>{otherParams.release_date}</Text>
                        <Text style={styles.popularityTextStyle}>{otherParams.popularity}</Text>
                        <View style={{flex: 1,justifyContent: 'flex-end'}}>
                            <Text style={styles.averageVoteTextStyle}>{otherParams.vote_average}/10</Text>
                        
                            <TouchableOpacity style={{ backgroundColor: '#5D5450', height: 50, padding: '10%', justifyContent: 'center'}}>
                                <Text style={{ color: '#fff', textAlign: 'center' }}>Add to Favourite</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                </View>
                <Text style={styles.overviewStyle}>{otherParams.overview}</Text>
                
                <Text style={{paddingTop: '10%', color: '#5D5450', fontWeight: '700' }}>TRAILERS</Text>
                <Separator />
                <TouchableOpacity style={styles.trailerBtnStyle}>
                        <Feather name="play-circle" size={24} color="#5D5450" />
                        <Text style={{ color: '#5D5450'}}>Play trailer 1</Text>
                </TouchableOpacity>
                <TouchableOpacity style={styles.trailerBtnStyle}>
                        <Feather name="play-circle" size={24} color="#5D5450" />
                        <Text style={{ color: '#5D5450'}}>Play trailer 2</Text>
                </TouchableOpacity>

            </View>
        </View>
    );
}


const styles = StyleSheet.create({
    topBarStyle: {
        height: '10%', 
        backgroundColor: '#5D5450', 
        justifyContent: 'center',
        padding: '5%'
    },
    titleStyle: {
        color: 'white',
        fontSize: 24,
        fontWeight: '900'
    }, 
    movieDetailStyle: {
        flexDirection: 'row', 
        // padding: '5%'
    },
    releaseDateTextStyle: {
        fontSize: 20
    },
    popularityTextStyle: {
        fontSize: 15,
        fontStyle: 'italic',
    },
    averageVoteTextStyle: {
        fontSize: 15,
        fontWeight: '500',
        paddingBottom: 30
    },
    overviewStyle: {
        paddingTop: 10
    },
    trailerBtnStyle: {
        flexDirection: 'row',
        backgroundColor: '#E1E2F5', 
        height: 55, 
        justifyContent: 'center',
        margin: 5, 
        alignItems: 'center',
    },
    
    separator: {
        //  marginVertical: 8,
        paddingVertical: '2%',
        borderBottomColor: '#737373',
        borderBottomWidth: StyleSheet.hairlineWidth,

      },
});

export default Details;
