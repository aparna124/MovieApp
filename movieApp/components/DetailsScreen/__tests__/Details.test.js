import React from 'react';
import renderer from 'react-test-renderer';
import Details from '../Details';

describe('<Details />', () => {
    test('details screen renders correctly', () => {
        const mockedParams = {
            route: { params: { movieId: '1', otherParams: {original_title: 'title', poster_path: 'url', vote_average: '8.1', release_date: '2022-09-02', overviewposter_path: 'path'}} },
            navigation: ''
          };
          const tree = renderer.create(<Details {...mockedParams} />).toJSON();
          expect(tree).toMatchSnapshot();
    })

    
})