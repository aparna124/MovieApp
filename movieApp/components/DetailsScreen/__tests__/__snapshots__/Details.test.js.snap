// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Details /> details screen renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#5D5450",
        "height": "10%",
        "justifyContent": "center",
        "padding": "5%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "fontWeight": "900",
        }
      }
    >
      title
    </Text>
  </View>
  <View
    style={
      Object {
        "padding": "5%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://image.tmdb.org/t/p/w185/url",
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "height": 250,
              "marginVertical": 5,
              "shadowColor": "#051934",
              "width": 355,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          2022-09-02
        </Text>
        <Text
          style={
            Object {
              "fontSize": 15,
              "fontStyle": "italic",
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "500",
                "paddingBottom": 30,
              }
            }
          >
            8.1
            /10
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#5D5450",
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
                "padding": "10%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "textAlign": "center",
                }
              }
            >
              Add to Favourite
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "paddingTop": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#5D5450",
          "fontWeight": "700",
          "paddingTop": "10%",
        }
      }
    >
      TRAILERS
    </Text>
    <View
      style={
        Object {
          "borderBottomColor": "#737373",
          "borderBottomWidth": 0.5,
          "paddingVertical": "2%",
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E1E2F5",
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#5D5450",
          }
        }
      >
        Play trailer 1
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E1E2F5",
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "center",
          "margin": 5,
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#5D5450",
          }
        }
      >
        Play trailer 2
      </Text>
    </View>
  </View>
</View>
`;
